libs/BLEKit/tests/CoreGattServerEventHandler_test.cpp

```diff
@@ -0,0 +1,154 @@
+	{
+		core_gatt_server_event_handler.registerServerWrite(mock_server_write_func.AsStdFunction());
+		core_gatt_server_event_handler.setServices(_services);
+	}
+	void TearDown() override { ble::delete_mocks(); }
+
+	BLE &ble = BLE::Instance();
+	CoreGattServerEventHandler core_gatt_server_event_handler {};
+
+	MockFunction<updateServiceFunction> mock_server_write_func;
+
+	const UUID _uuid {0x0001};
+	static const uint8_t _services_count {1};
+	static const uint8_t _characteristics_count {1};
+	uint8_t _characteristic_value {42};
+	ReadOnlyGattCharacteristic<uint8_t> _writable_characteristic {0x1234, &_characteristic_value};
+	std::array<GattCharacteristic *, _characteristics_count> _characteristic_table {&_writable_characteristic};
+	mock::BLEService mock_service = mock::BLEService(_uuid, _characteristic_table, _characteristics_count);
+
+	std::array<interface::BLEService *, _services_count> _services {&mock_service};
+};
+
+MATCHER_P(compareUpdateServiceFunction, expected_function, "")
+{
+	bool same_target_type = expected_function.target_type().name() == arg.target_type().name();
+
+	return same_target_type;
+}
+
+MATCHER_P(compareGattWriteCallbackParams, expected_params, "")
+{
+	bool same_handle = expected_params.handle == arg.handle;
+
+	return same_handle;
+}
+
+TEST_F(CoreGattServerEventHandlerTest, initialization)
+{
+	EXPECT_NE(&core_gatt_server_event_handler, nullptr);
+}
+
+TEST_F(CoreGattServerEventHandlerTest, registerServerWrite)
+{
+	updateServiceFunction dummy_function = [](GattAttribute::Handle_t, const uint8_t *, uint16_t) { return; };
+	std::function<void()> impostor_function;
+
+	core_gatt_server_event_handler.registerServerWrite(dummy_function);
+
+	EXPECT_CALL(mock_service, getCharacteristicCount).WillOnce(Return(_characteristics_count));
+
+	EXPECT_CALL(mock_service, updateData(compareUpdateServiceFunction(dummy_function))).Times(_characteristics_count);
+	EXPECT_CALL(mock_service, updateData(compareUpdateServiceFunction(impostor_function))).Times(0);
+
+	core_gatt_server_event_handler.updateData();
+}
+
+TEST_F(CoreGattServerEventHandlerTest, setServices)
+{
+	const UUID uuid {0x0001};
+	static const uint8_t serivce_1_characteristics_count {1};
+	static const uint8_t serivce_2_characteristics_count {2};
+	static const uint8_t services_count {2};
+	uint8_t characteristic_value {};
+	ReadOnlyGattCharacteristic<uint8_t> writable_characteristic {0x1234, &characteristic_value};
+	std::array<GattCharacteristic *, serivce_1_characteristics_count> service_1_characteristic_table {
+		&writable_characteristic};
+	std::array<GattCharacteristic *, serivce_2_characteristics_count> service_2_characteristic_table {
+		&writable_characteristic, &writable_characteristic};
+	mock::BLEService mock_service_1 =
+		mock::BLEService(uuid, service_1_characteristic_table, serivce_1_characteristics_count);
+	mock::BLEService mock_service_2 =
+		mock::BLEService(uuid, service_2_characteristic_table, serivce_2_characteristics_count);
```

Comment:
c'est très dur à lire et à suivre.

il faut mettre un peu plus d'`auto` et `std::to_array` et raccourcir le nom des variables.

```suggestion
	ReadOnlyGattCharacteristic<uint8_t> charac {0x1234, 0};
	auto s1_charac_table = std::to_array<GattCharacteristic *>({&charac}); // ? One characteristic
	auto s2_charac_table = std::to_array<GattCharacteristic *>({&charac, &charac}); // ? Two characteristics
	
	const UUID s1_uuid {0x0001};
	auto s1 = mock::BLEService(s1_uuid, s1_charac_table, std::size(s1_charac_table);
	const UUID s2_uuid {0x0002};
	auto s1 = mock::BLEService(s2_uuid, s2_charac_table, std::size(s2_charac_table);
```